WordPress/Classes/ViewRelated/Reader/ReaderHelpers.swift

```diff
@@ -233,3 +233,43 @@ import WordPressShared
         return AccountHelper.isDotcomAvailable()
     }
 }
+
+/// Reader tab items
+extension ReaderHelpers {
+    /// Display title for topics that do not use the default title
+    class func displayTitle(for topic: ReaderAbstractTopic) -> String {
+        if topicIsFollowing(topic) {
+            return NSLocalizedString("Following", comment: "Title of the Following Reader tab")
+        } else if topicIsLiked(topic) {
+            return NSLocalizedString("Likes", comment: "Title of the Likes Reader tab")
+        }
+        return topic.title
+    }
+    /// Sorts the default tabs according to the order [Following, Discover, Likes], and adds the Saved tab
+    class func rearrange(items: [ReaderTabItem]) -> [ReaderTabItem] {
+        var mutableItems = items
+        mutableItems.sort {
```

Comment:
I agree that is not super readable; I also agree that 1 could be tricky. I have added some comments in that method, hoping that they add some clarity to it.

Reply:
I think this may be more refactoring than necessary for this PR, but here are some thoughts I had looking through this:

It seems like we could replace these individual methods with an enum for each topic type:

```swift
enum ReaderTopicType {
    case following
    case discover
    case liked
    
    var path: String {
        switch self {
        case .following:
            return "/read/following"
        case .discover:
            return "/read/sites/53424024/posts"
        case .liked:
            return "/read/liked"
        }
    }
}
```

The sort code could then take a list of topics for sorting and be more general. Although, this might be harder to reason about in the end.

```swift
let sortedTopics: [ReaderTopicType] = [.following, .discover, .liked]

var mutableItems = items

mutableItems.sort { (leftTopic, rightTopic) in
    let lhsIndex = sortedTopics.firstIndex { type in
        return leftTopic.topic?.path.hasSuffix(type.path) == true
    }
    let rhsIndex = sortedTopics.firstIndex { type in
        return rightTopic.topic?.path.hasSuffix(type.path) == true
    }
    
    if let lhsIndex = lhsIndex, let rhsIndex = rhsIndex {
        /// Sort from `sortedTopics`
        return lhsIndex < rhsIndex
    } else if lhsIndex != nil || rhsIndex != nil {
        /// Sort any topics from`sortedTopics` over those not in the list
        return lhsIndex != nil
    } else if leftTopic.topic?.type == rightTopic.topic?.type {
        /// Sort any others topics alphabetically, grouped by topic type
        return leftTopic.title < rightTopic.title
    } else {
        return true
    }
}

// fourth item: Saved. It's manually inserted after the sorting
mutableItems.insert(ReaderTabItem(title: NSLocalizedString("Saved", comment: "Title of the Saved Reader Tab")), at: 3)
return mutableItems
```

Reply:
This is a valid point, and I also agree that this kind of refactoring could go in a separate PR. Another thought that I had is that, if the default topics are not going to change for the foreseeable future, maybe we don't even need to fetch them? If we all agree, I'd address this task on a separate PR, once the main navigation is in place.